apps/admin/backend/src/server.ts

```diff
@@ -76,9 +82,16 @@ export async function start({
       config: {
         allowElectionManagersToAccessUnconfiguredMachines: false,
       },
-      logger,
+      logger: baseLogger,
     });
 
+    const logger = Logger.from(baseLogger, () =>
+      getUserRole(auth, assertDefined(resolvedWorkspace))
+    );
```

Comment:
Another way you could work around the bootstrapping problem:

```ts
// Start with a Logger that always returns `system`
const systemLogger = new Logger(LogSource.VxAdminService, () => 'system');

// Once auth is defined, make another Logger with auth
const appLogger = new Logger(LogSource.VxAdminService, () => getUserRole(auth, resolvedWorkspace))
```

I think the advantage would be avoiding the confusion of having to understand the difference between `BaseLogger` and `Logger` when reading this file.

Reply:
`libs/auth` is sometimes logging with user roles, like for card programming, it's logging as `system_administrator` because it knows that's the only role that can program cards.

And I think that the `app` could have use cases where it wants to log as `system`, which is distinct from `unknown`, so I still see a need for the `Logger` to have two methods rather than just one `log` method.

Reply:
Gotcha. That all makes sense. I'm not suggesting we get rid of `BaseLogger` altogether, just not using it for this particular use case. What do you think of that idea?

Reply:
OK, I think I see what you mean. You mean, not starting with `BaseLogger`, but rather with a systemLogger so all the loggers you see are `Logger`? I like that idea. I might change `logAsCurrentUser` to `logAsCurrentRole` in that case. And in your example,  I would want to make `appLogger` with `Logger.from` for testing purposes. How does that sound?

Reply:
Ah, I didn't think about the testing needs for using `Logger.from`. A comment might help explain that pattern, since it's a little confusing.

Reply:
I'm still feeling fuzzy on what the best pattern here is so I'm going to leave it for now and re-evaluate once logging's pretty exclusively on the backend.